tasks/upgrade.yaml

```diff
@@ -0,0 +1,95 @@
+        default: ""
+
+      bundle_path:
+        description: Path relative to the repositories root where the bundle needs to go
+        default: bundle
+
+      path:
+        description: Path relative to the repositories root where the package needs to go
+        default: .
+
+      team:
+        description: The team you want to pull the package from
+        default: uds
+
+      target_repo:
+        description: The repository to pull from
+        default: ghcr.io/defenseunicorns/packages
+
+      architecture:
+        description: The architecture of the package to create
+        default: ${UDS_ARCH}
+
+    actions:
+      - task: utils:determine-repo
+        with:
+          team: ${{.inputs.team}}
+
+      - description: Remove cloned repo
+        cmd: rm -rf upgrade-test
+
+      - description: Clone latest tag
+        cmd: git clone --depth 1 --branch $(git describe --tags `git rev-list --tags --max-count=1`) $(git remote get-url origin) upgrade-test
+
+      - description: Create test bundle
+        shell:
+          linux: bash
+          darwin: bash
+        cmd: |
+          cd upgrade-test
+
+          FLAVOR_CHECK="flavor: ${FLAVOR}"
+
+          # Checks if flavor exists
+          if grep -q "${FLAVOR_CHECK}" ./zarf.yaml; then
+
+            # VARIABLES
+            PACKAGE_NAME=$(cat ${{ .inputs.path }}/zarf.yaml | yq .metadata.name)
+            LATEST_VERSION=$(./uds zarf tools registry ls ${TARGET_REPO}/${PACKAGE_NAME} | grep ${FLAVOR} | sort -V | tail -1)
+            BUNDLE_NAME=$(cat ${{ .inputs.bundle_path }}/uds-bundle.yaml | ./uds zarf tools yq .metadata.name)
+            BUNDLE_VERSION=$(cat ${{ .inputs.bundle_path }}/uds-bundle.yaml | ./uds zarf tools yq .metadata.version)
+
+            # Pulls latest published zarf package
+            ./uds zarf package pull \
+              oci://${TARGET_REPO}/${PACKAGE_NAME}:${LATEST_VERSION} \
+              --no-progress \
+              -o ${{ .inputs.path }}
+
+            # Checks if the zarf package is registry1 and ARM
+            if [ ${FLAVOR} != "registry1" ] || [ ${{ .inputs.architecture }} != "arm64" ]; then
+
+              # Creates and moves the bundle into the correct spot
+              ./uds create ${{ .inputs.bundle_path }} \
+                --architecture=${{ .inputs.architecture }} \
+                --confirm \
+                --no-progress \
+                ${{ .inputs.options }}
+
+              mv ${{ .inputs.bundle_path }}/uds-bundle-${BUNDLE_NAME}-${{ .inputs.architecture }}-${BUNDLE_VERSION}.tar.zst ../bundle
+            else
+              echo "Registry1 bundles cannot be made for 'arm64'"
+            fi
+
+            cd ..
+
+            ./uds run test-upgrade --no-progress --set FLAVOR=${FLAVOR} ${{ .inputs.options }} --no-progress
+
+          # If flavor doesn't exist
+          else
+            cd ..
+            echo "::warning::⚠️ Latest tag does not have flavor:${FLAVOR}"
```

Comment:
This is possible to show in GH with callable workflows: https://github.com/defenseunicorns/uds-common/pull/211

Reply:
Doing so would also avoid the case where this task fails because the repo is new and has no tags.